server/image_list.go

```diff
@@ -23,9 +26,15 @@ func (s *Server) ListImages(ctx context.Context, req *types.ListImagesRequest) (
 	if err != nil {
 		return nil, err
 	}
+	// exclude pinned images from the results
+	filteredImagesList, err := FilterPinnedImages(results, s.config.PinnedImages)
+	if err != nil {
+		return nil, err
+	}
+
 	resp := &types.ListImagesResponse{}
-	for i := range results {
-		image := ConvertImage(&results[i])
+	for i := range filteredImagesList {
+		image := ConvertImage(&filteredImagesList[i])
```

Comment:
CRI has a [pinned](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto#L1404) field already. I think  we should continue to show the kubelet the pinned images but make sure we add the pinned field to them